packages/updatePrescriptionStatus/src/updatePrescriptionStatus.ts

```diff
@@ -0,0 +1,217 @@
+          details: {
+            coding: [
+              {
+                system: "https://fhir.nhs.uk/CodeSystem/http-error-codes",
+                code: "BAD_REQUEST",
+                display: "400: The Server was unable to process the request."
+              }
+            ]
+          }
+        }
+      ]
+    }
+    return {
+      statusCode: 400,
+      body: JSON.stringify(errorResponseBody),
+      headers: {
+        "Content-Type": "application/fhir+json",
+        "Cache-Control": "no-cache"
+      }
+    }
+  }
+
+  const entries = requestBody.entry
+
+  if (!entries || entries.length === 0) {
+    logger.error("Missing required fields")
+    const errorResponseBody = {
+      resourceType: "OperationOutcome",
+      issue: [
+        {
+          code: "value",
+          severity: "error",
+          details: {
+            coding: [
+              {
+                system: "https://fhir.nhs.uk/CodeSystem/Spine-ErrorOrWarningCode",
+                code: "MISSING_VALUE",
+                display: "Missing required fields"
+              }
+            ]
+          }
+        }
+      ]
+    }
+    return {
+      statusCode: 400,
+      body: JSON.stringify(errorResponseBody),
+      headers: {
+        "Content-Type": "application/fhir+json",
+        "Cache-Control": "no-cache"
+      }
+    }
+  }
+
+  const responseBundle: any = {
+    resourceType: "Bundle",
+    id: uuidv4(),
+    meta: {
+      lastUpdated: new Date().toISOString()
+    },
+    type: "transaction-response",
+    entry: []
+  }
+
+  for (const entry of entries) {
+    logger.info("Processing entry", {entry: entry})
+    const entry_resource = entry.resource
+
+    const dynamoDBItem: DynamoDBItem = {
+      RequestID: xRequestId,
+      PrescriptionID: entry_resource.basedOn?.[0]?.identifier?.value,
+      PatientNHSNumber: entry_resource.for?.identifier?.value,
+      PharmacyODSCode: entry_resource.owner?.identifier?.value,
+      TaskID: entry_resource.id,
+      LineItemID: entry_resource.focus?.identifier?.value,
+      TerminalStatusIndicator: entry_resource.status,
+      LastModified: entry_resource.lastModified,
+      Timestamp: new Date().toISOString(),
+      RequestMessage: entry_resource
+    }
+
+    const invalidFields = []
+    for (const [field, value] of Object.entries(dynamoDBItem)) {
+      if (!value) {
+        logger.info("Invalid value", {field: field, value: value})
+        invalidFields.push(field)
+      }
+    }
+
+    if (invalidFields.length > 0) {
+      const errorMessage = `Missing required fields: ${invalidFields.join(", ")}`
+      logger.error("Error message", {errorMessage: errorMessage})
+      const errorResponseBody = {
+        resourceType: "OperationOutcome",
+        issue: [
+          {
+            code: "value",
+            severity: "error",
+            details: {
+              coding: [
+                {
+                  system: "https://fhir.nhs.uk/CodeSystem/Spine-ErrorOrWarningCode",
```

Comment:
Same as a previous error, this should probably be a http-error-code

Reply:
The code system has been changed.